ledger/participant-integration-api/src/main/scala/platform/store/appendonlydao/events/BufferedTransactionsReader.scala

```diff
@@ -180,19 +186,31 @@ private[platform] object BufferedTransactionsReader {
       totalRetrievedCounter: Counter,
       outputStreamBufferSize: Int,
       bufferSizeCounter: Counter,
+      inStreamBufferLength: Counter,
   )(implicit executionContext: ExecutionContext): Source[(Offset, API_RESPONSE), NotUsed] = {
     def bufferedSource(
         slice: Vector[(Offset, TransactionLogUpdate)]
     ): Source[(Offset, API_RESPONSE), NotUsed] =
       Source
-        .fromIterator(() => slice.iterator)
-        // Using collect + mapAsync as an alternative to the non-existent collectAsync
-        .collect { case (offset, tx: TxUpdate) =>
-          Timed.future(toApiTxTimer, toApiTx(tx, filter, verbose).map(offset -> _))
+        .fromIterator(() =>
+          slice
+            .collect { case (offset, tx: TransactionLogUpdate.TransactionAccepted) =>
+              offset -> tx
+            }
+            .iterator
+            .grouped(10)
```

Comment:
Bit magical

Reply:
Missed this one. The purpose here was to reduce Futures/Akka tripping. Good observation, it won't make it like this int the prod PR.